### PR TITLE
Add more context to errors in legacy wat parser

### PR DESCRIPTION
**Pull request by tlively**
To make the error messages more useful, especially when parsing extremely large
wat files, print the erroneous fragments along with the existing error messages.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5980** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5980" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5980?utm_source=stack-comment).

File: src/wasm/wasm-s-parser.cpp [hunk 1]
```diff
@@ -516,7 +524,8 @@ Name SExpressionWasmBuilder::getElemSegmentName(Element& s) {
     // index
     size_t offset = parseIndex(s);
     if (offset >= elemSegmentNames.size()) {
-      throw ParseException("unknown elem segment", s.line, s.col);
+      throw ParseException(
+        "unknown elem segment"s + s.toString(), s.line, s.col);
```
Comment (kripken) [inline src/wasm/wasm-s-parser.cpp:528]:
```suggestion
        "unknown elem segment: "s + s.toString(), s.line, s.col);
```

File: src/wasm/wasm-s-parser.cpp [hunk 2]
```diff
@@ -503,7 +510,8 @@ Name SExpressionWasmBuilder::getTableName(Element& s) {
     // index
     size_t offset = parseIndex(s);
     if (offset >= tableNames.size()) {
-      throw ParseException("unknown table in getTableName", s.line, s.col);
+      throw ParseException(
+        "unknown table in getTableName"s + s.toString(), s.line, s.col);
```
Comment (kripken) [inline src/wasm/wasm-s-parser.cpp:514]:
```suggestion
        "unknown table in getTableName: "s + s.toString(), s.line, s.col);
```

**Review by kripken — COMMENTED** [review]
Would it be simpler perhaps to give `ParseException` the `Element& s` as a parameter, instead of `line,col`, and then it could append both the text and the `line,col`?

**Comment by tlively** [comment]
> Would it be simpler perhaps to give `ParseException` the `Element& s` as a parameter, instead of `line,col`, and then it could append both the text and the `line,col`?

On the one hand, yes, but on the other hand, I don't want to touch this EOL code more 😢

**Comment by kripken** [comment]
Would you like me to look into it? I wouldn't mind.

**Comment by kripken** [comment]
Quick sketch of what I mean: https://gist.github.com/kripken/a8493f8944e2fc5d8882fdff92f55e22

Happy to fill that out into a PR, if you don't object.

**Comment by tlively** [comment]
Perfect, thanks, that would be great.

**Pull request closed without merging**